torch/csrc/jit/ivalue.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+#include <ATen/core/ivalue.h>
+
+namespace torch {
+namespace jit {
+
+using ::c10::ivalue::List;
+using ::c10::ivalue::Shared;
+
+using ::c10::IValue;
+using ::c10::ivalue::Future;
+using ::c10::ivalue::Tuple;
+
+using ::c10::ivalue::BoolList;
+using ::c10::ivalue::DoubleList;
+using ::c10::ivalue::GenericList;
+using ::c10::ivalue::IntList;
+using ::c10::ivalue::TensorList;
+
+using ::c10::ivalue::ConstantString;
+
+// XXX: This function is to specialize IValue for tensor type in
+// interpreter, it should only be used in JIT and not anywhere
+// else, that's why we keep this in JIT but not in c10::ivalue
```

Comment:
I added back torch/csrc/jit/ivalue.h because we need to specialize the undefined tensor semantic for the JIT interpreter, and we don't want to put this `toOptionalTensor` function to c10 to avoid improper usages in any other spaces. We can remove this again in the future when we get rid of undefined tensor in TH. CC @ZolotukhinM @zdevito

Reply:
Can we just define `toOptionalTensor` where it's used? I think there is only one place? Is there expected to be more places?

Reply:
This function is specific to the ATen bindings to the interpreter and should live there.

Reply:
Also, this file was deleted in a previous commit and this rebase messed it up. It should remain deleted.